src/inmanta/server/services/orchestrationservice.py

```diff
@@ -814,9 +820,163 @@ async def _put_version(
             all_agents: set[str] = {res.agent for res in rid_to_resource.values()}
             all_agents.add(const.AGENT_SCHEDULER_ID)
 
+            type_to_agent: dict[str, set[str]] = defaultdict(set)
+            for res in rid_to_resource.values():
+                type_to_agent[str(res.resource_type)].add(res.agent)
+
             for agent in all_agents:
                 await self.agentmanager_service.ensure_agent_registered(env, agent, connection=connection)
 
+            async def register_code(
+                cm_version: int,
+                environment: uuid.UUID,
+                module_version_info: dict[str, dict[str, Any]],
+                type_to_module_data: dict[str, set[str]],
+                type_to_agent: dict[str, set[str]],
+                connection: Connection,
+            ) -> None:
+
+                query = """
+                    INSERT INTO modules_for_agent(
+                        cm_version,
+                        environment,
+                        agent_name,
+                        inmanta_module_name,
+                        inmanta_module_version
+                    ) VALUES(
+                        $1,
+                        $2,
+                        $3,
+                        $4,
+                        $5
+                    )
+                    ON CONFLICT DO NOTHING;
+                """
+                async with connection.transaction():
+                    values = []
+                    for resource_type in type_to_agent.keys():
+                        for agent_name in type_to_agent[resource_type]:
+                            if resource_type not in type_to_module_data:
+                                LOGGER.debug("No inmanta module information was provided for resource type %s." % resource_type)
+                                continue
+                            for module_name in type_to_module_data[resource_type]:
+                                values.append(
+                                    (
+                                        cm_version,
+                                        environment,
+                                        agent_name,
+                                        module_name,
+                                        module_version_info[module_name]["version"],
+                                    )
+                                )
+
+                    await connection.executemany(
+                        query,
+                        values,
+                    )
```

Comment:
this should go into the data package I think (i.e. in the relevant objects)

I think this applies to all queries